samtranslator/model/api/http_api_generator.py

```diff
@@ -397,34 +397,45 @@ def _construct_alias_target(self, domain):
 
     def _add_auth(self):
         """
-        Add Auth configuration to the OAS file, if necessary
+        Add Auth configuration to the OAS file. In order to support the built-in AWS_IAM authorizer it is always added.
         """
-        if not self.auth:
-            return
-
-        if self.auth and not self.definition_body:
+        if not self.definition_body:
             raise InvalidResourceException(
                 self.logical_id, "Auth works only with inline OpenApi specified in the 'DefinitionBody' property."
             )
 
-        # Make sure keys in the dict are recognized
-        if not all(key in AuthProperties._fields for key in self.auth.keys()):
-            raise InvalidResourceException(self.logical_id, "Invalid value for 'Auth' property")
-
         if not OpenApiEditor.is_valid(self.definition_body):
             raise InvalidResourceException(
                 self.logical_id,
                 "Unable to add Auth configuration because 'DefinitionBody' does not contain a valid OpenApi definition.",
             )
+
         open_api_editor = OpenApiEditor(self.definition_body)
-        auth_properties = AuthProperties(**self.auth)
-        authorizers = self._get_authorizers(auth_properties.Authorizers, auth_properties.DefaultAuthorizer)
 
-        # authorizers is guaranteed to return a value or raise an exception
-        open_api_editor.add_authorizers_security_definitions(authorizers)
-        self._set_default_authorizer(
-            open_api_editor, authorizers, auth_properties.DefaultAuthorizer, auth_properties.Authorizers
-        )
+        # To remain backwards compatible add the built-in "AWS_IAM" security scheme _before_ adding the authorizers defined
+        # in the template so that if the template already has an authorized named "AWS_IAM" it will override the built-in one.
```

Comment:
Why wouldn't we add this only if a customer wants AWS_IAM auth?

Reply:
The goal here is that if a customer somehow already has an authorizer called `AWS_IAM` we wouldn't want to clobber it.